docs/source/performance/caching.md

```diff
@@ -68,80 +119,142 @@ const resolvers = {
+
+```js
+const server = new ApolloServer({
+  // ...other options...
+  cacheControl: {
+    calculateHttpHeaders: false,
+  },
+}));
+```
 
-For any response whose overall cache policy has a non-zero `maxAge`, Apollo Server will automatically set the `Cache-Control` HTTP response header to an appropriate value describing the `maxAge` and scope, such as `Cache-Control: max-age=60, private`.  If you run your Apollo Server instance behind a [CDN](https://en.wikipedia.org/wiki/Content_delivery_network) or other caching proxy, it can use this header's value to know how to cache your GraphQL responses.
 
-As many CDNs and caching proxies only cache GET requests (not POST requests) and may have a limit on the size of a GET URL, you may find it helpful to use [automatic persisted queries](https://github.com/apollographql/apollo-link-persisted-queries), especially with the `useGETForHashedQueries` option to `apollo-link-persisted-queries`.
 
-If you don't want to set HTTP cache headers, pass `cacheControl: {calculateHttpHeaders: false}` to `new ApolloServer()`.
+## Caching with `responseCachePlugin` (advanced)
 
-## Saving full responses to a cache
+You can cache Apollo Server query responses in stores like Redis, Memcached, or Apollo Server's in-memory cache.
 
-Apollo Server lets you save cacheable responses to a Redis, Memcached, or in-process cache. Cached responses respect the `maxAge` cache hint.
+### In-memory cache setup
 
-To use the response cache, you need to install its plugin when you create your `ApolloServer`:
+To set up your in-memory response cache, you first import the `responseCachePlugin` and provide it to the `ApolloServer` constructor:
 
 ```javascript
 import responseCachePlugin from 'apollo-server-plugin-response-cache';
+
 const server = new ApolloServer({
-  // ...
+  // ...other options...
   plugins: [responseCachePlugin()],
 });
 ```
 
-By default, the response cache plugin will use the same cache used by other Apollo Server features, which defaults to an in-memory LRU cache. When running multiple server instances, you’ll want to use a shared cache backend such as Memcached or Redis instead.  See [the data sources documentation](/data/data-sources/#using-memcachedredis-as-a-cache-storage-backend) for details on how to customize Apollo Server's cache.  If you want to use a different cache backend for the response cache than for other Apollo Server caching features, just pass a `KeyValueCache` as the `cache` option to the `responseCachePlugin` function.
+On initialization, this plugin automatically begins caching responses according to [field settings](#in-your-schema-static).
+
+The plugin uses the same in-memory LRU cache as Apollo Server's other features. For environments with multiple server instances, you should instead use a shared cache backend, such as [Memcached or Redis](#memcachedredis-setup).
```

Comment:
I might say _might_ instead of _should_ use a shared cache backend, but offer the benefits of doing so (cached once per cluster).  In many cases, the overhead of operating a network cache could easily outweigh the benefits of just caching them on even an entire fleet and this really depends on use-case.  Those who already have one operating, would certainly be tempted/drawn to such an option, but I wouldn't want it to preclude people from trying this on account of complexity!